go/str_pad.go

```diff
@@ -0,0 +1,94 @@
+package pehape
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+)
+
+const (
+	STR_PAD_RIGHT int = iota
+	STR_PAD_LEFT
+	STR_PAD_BOTH
+)
+
+var (
+	ErrTooManyArgs           = errors.New("too many args")
+	ErrPadTypeInvalid        = errors.New("invalid pad type")
+	ErrPadInvalid            = errors.New("invalid pad")
+	ErrPadMustNotEmptyString = errors.New("pad must be a non-empty string")
+)
+
+// StrPad Pad a string to a certain length with another string
+// This function returns the string string padded on the left, the right,
+// or both sides to the specified padding length.
+// If the optional argument pad_string is not supplied, the string is padded with spaces,
+// otherwise it is padded with characters from pad_string up to the limit.
+// Optional params:
+// - pad_string => string or int or float64
+// - pad_type => STR_PAD_LEFT, STR_PAD_RIGHT, STR_PAD_BOTH
+func StrPad(s string, length int, args ...any) (string, error) {
+	// first check args length
+	if len(args) > 2 {
+		return "", ErrTooManyArgs
+	}
+
+	// check if length < len(s)
+	if length < len(s) {
+		return s, nil
+	}
+
+	padString := " "
+
+	// set pad
+	if len(args) > 0 {
+		switch args[0].(type) {
+		case string, float64, int:
+			padString = fmt.Sprintf("%v", args[0])
+		default:
+			return "", ErrPadInvalid
+		}
+
+		// validate pad only when args[0] not empty
+		if padString == "" {
+			return "", ErrPadMustNotEmptyString
+		}
+	}
+
+	// set pad type
+	var padType int
+	if len(args) == 2 {
+		switch args[1].(type) {
+		case int:
+			padType = args[1].(int)
+		case float64:
+			padType = int(args[1].(float64))
+		default:
+			return "", ErrPadTypeInvalid
+		}
```

Comment:
According to the `str_pad` [documentation here](https://www.php.net/manual/en/function.str-pad.php), `padType` can also be a string. Try:
```php
<?php
echo str_pad("Alien", 12, " ", "1");
```